File: core/src/main/java/org/toml/ide/experiments/TomlExperiments.java
/*
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package org.toml.ide.experiments;

import consulo.application.util.registry.Registry;

public final class TomlExperiments {
    public static final String JSON_SCHEMA = "org.toml.json.schema";

    private TomlExperiments() {
    }

    public static boolean isJsonSchemaEnabled() {
        return isFeatureEnabled(JSON_SCHEMA);
    }

    private static boolean isFeatureEnabled(String registryKey) {
        return Registry.is(registryKey, false);
    }
}

File: core/src/main/kotlin/org/toml/ide/experiments/TomlExperiments.kt
/*
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package org.toml.ide.experiments

import consulo.application.util.registry.Registry

object TomlExperiments {
    const val JSON_SCHEMA = "org.toml.json.schema"

    val isJsonSchemaEnabled: Boolean
      get() = isFeatureEnabled(JSON_SCHEMA)

    @Suppress("SameParameterValue")
    private fun isFeatureEnabled(registryKey: String): Boolean =
        Registry.`is`(registryKey, false)
}
